fieldservice_maintenance/models/maintenance_request.py

```diff
@@ -19,12 +20,18 @@ def create(self, vals):
             fsm_equipment = self.env['fsm.equipment'].search(
                 [('maintenance_equipment_id', '=', request.equipment_id.id)],
                 limit=1)
-            fsm_order_id = self.env['fsm.order'].create(
-                {'type': 'maintenance',
-                 'equipment_id': fsm_equipment.id,
-                 'location_id': fsm_equipment and
-                 fsm_equipment.current_location_id.id,
-                 'request_id': request.id
-                 })
-            request.fsm_order_id = fsm_order_id
+            fsm_order_type = self.env['fsm.order.type'].search(
+                [('internal_type', '=', 'maintenance')],
+                order="id desc", limit=1)
+            try:
+                fsm_order_id = self.env['fsm.order'].create(
+                    {'type': fsm_order_type.id,
+                     'equipment_id': fsm_equipment.id,
+                     'location_id': fsm_equipment and
+                     fsm_equipment.current_location_id.id,
+                     'request_id': request.id
+                     })
+                request.fsm_order_id = fsm_order_id
+            except:
+                pass
```

Comment:
Silencing exceptions is a code smell.
Why do we need this?

Reply:
Done